Interaction/Widgets/vtkSeedWidget.cxx

```diff
@@ -216,10 +236,42 @@ void vtkSeedWidget::CompletedAction(vtkAbstractWidget *w)
 {
   vtkSeedWidget *self = reinterpret_cast<vtkSeedWidget*>(w);
 
-  // Do something only if we are in the middle of placing the seeds
+  // If we are in the middle of placing the seeds, end the placing and return
   if ( self->WidgetState == vtkSeedWidget::PlacingSeeds )
   {
     self->CompleteInteraction();
+    return;
+  }
+
+  // Manipulate handle
+  if ( self->WidgetState == vtkSeedWidget::SelectedSeed )
+  {
+    return;
+  }
+
+  self->InvokeEvent(vtkCommand::MouseMoveEvent, nullptr);
+
+  // compute some info we need for all cases
+  int X = self->Interactor->GetEventPosition()[0];
+  int Y = self->Interactor->GetEventPosition()[1];
+
+  // When a seed is placed, a new handle widget must be created and enabled.
+  int state = self->WidgetRep->ComputeInteractionState(X,Y);
+  if ( state == vtkSeedRepresentation::NearSeed )
```

Comment:
Why this special case needs to be handled like this?
Do we simulate a left button press event here? What happens if left button press is not mapped to AddPoint?
Is it a good idea in general to hardcode self->SelectionButton = vtkSeedWidget::RightButton (will the accept it to be merged into VTK)?

Reply:
I simulate the selection (but I don't allow the dragging of the point, i.e. changing its position). In this way when the user apply right click (or middle click or any double click), he/she can see the point selected and therefore the user will have feedback of which one has been clicked (e.g., when you have very nearby points, this can be useful). See https://www.dropbox.com/s/h8rcisyejpjov0h/rightclick.mkv?dl=0

I see in general the problem of the idea of the self->SelectionButton. In fact the actions can be mapped to different vtkCommand (e.g. leftpressbutton to CompleteAction instead to AddPoint). And hardcoding for each Action a "SelectionButton" most likley will not be accepted in a VTK merge. The problem is that in Slicer I can observe only the Interaction (e.g., vtkCommand::StartInteractionEvent) for the vtkSeedWidget. For example if I observe in Slicer vtkCommand::LeftButtonPressEvent from a vtkSeedWidget this will never happen. 

I'll think a little bit more how to get that when the interaction of a vtkSeedWidget is finished was due to a left click or right click or etc...

Reply:
maybe I can implement that in the moment of the mapping of the vtkcommand to the vtkWidgetEvent and Action, also the "SelectionButton" will be mapped. So something like that:

vtkCommand::LeftButtonPressEvent - > vtkWidgetEvent::AddPoint -> vtkSeedWidget::AddPointAction -> vtkSeedWidget::LeftButton


vtkCommand::RightButtonReleaseEvent - > vtkWidgetEvent::AddPoint -> vtkSeedWidget::AddPointAction -> vtkSeedWidget::RightButton